### DISCUSSION TITLE
Any convenient way to set children default theme in parent control theme

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#8837] [Q&A]
Hello I have a customized `ListBox` wrapping customized `ListBoxItem`. The `ListBoxItem` default `ControlTheme` will vary according to `ListBox` properties. Is there anyway to achieve this?
Example scenario: Make ListBoxItem Margin vary according to ListBox Orientation. 
My current solution is to define several `ControlTemplates` in parent `ControlTheme.Resources`, with each defines a different `ListBoxItem` default `ControlTheme`. and set the `ControlTemplates` with different style selector.

**Comment by timunie** [comment]
You may also use a converter or a behavior.

**Reply by rabbitism** [reply]
Hmm, but I need to use a lot of dynamic resources. is it possible to return dynamic resource via converter?

**Reply by timunie** [reply]
Tbh I don't know your use case. So for sure, you can just use DynamicResource everywhere and just set the new Brushes where you need it. I think you cannot return DynamicResource from Converter, at least not easy.

**Comment by maxkatz6** [comment]
You might need ItemContainerTheme
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/Menu.xaml#L94
